utils/build-click.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash -e
+
+KAIDAN_SOURCES=$(dirname "$(readlink -f "${0}")")/..
+
+ARCH=$(dpkg-architecture -qDEB_HOST_ARCH)
+DEB_HOST_MULTIARCH=$(dpkg-architecture -qDEB_HOST_MULTIARCH)
+
+CLICK_TARGET_DIR="$KAIDAN_SOURCES/bin/ubuntu-touch/tmp/"
+
+! [ -d $CLICK_TARGET_DIR ] && mkdir $CLICK_TARGET_DIR
+
+
+echo "*****************************************"
+echo "Downloading Debian packages"
+echo "*****************************************"
+
+echo "Kirigami2"
+for PKG in qml-module-org-kde-kirigami2 kirigami2-dev libkf5kirigami2-5; do
+	VERSION="5.43.0+p16.04+git20180221.0430-0"
+	curl --progress-bar https://archive.neon.kde.org/dev/unstable/pool/main/k/kirigami2/${PKG}_${VERSION}_$ARCH.deb -o /tmp/${PKG}_${VERSION}_$ARCH.deb
+	dpkg-deb -x /tmp/${PKG}_${VERSION}_$ARCH.deb $CLICK_TARGET_DIR
+	rm /tmp/${PKG}_${VERSION}_$ARCH.deb
```

Comment:
Please create a function for this, so there are not three copies of the same code in here.
E.g.: `install_deb https://archive.neon.kde.org/dev/unstable/pool/main/k/kirigami2 ${PKG} ${KIRIGAMI_VERSION}`

And maybe declare the versions at the top of the file (`KIRIGAMI_VERSION="5.43.0+p16.04+git20180221.0430-0"`, ...)

Reply:
And you could also move this function to the travis common file (it could be used there, too)

Reply:
I don't think that this is **much more** text:
```bash
# UNTESTED!
install_deb() {
    BASE_URL="${1}"; PKG="${2}"; VERSION="${3}"
    DEB_NAME="${PKG}_${VERSION}_${ARCH}.deb"

    # download deb using curl with a nice progress bar
    curl --progress-bar ${BASE_URL}/${DEB_NAME} -o /tmp/${DEB_NAME}

    # install either in click target folder or in the running system
    if [[ ${PLATFORM} == "click" ]]; then
        dpkg-deb -x /tmp/${DEB_NAME} ${CLICK_TARGET_DIR}
    else
        sudo dpkg -i /tmp/${DEB_NAME} || sudo apt-get -f -y install
    fi

    # clean up
    rm /tmp/${DEB_NAME}
}

for PKG in qml-module-org-kde-kirigami2 kirigami2-dev libkf5kirigami2-5; do
    install_deb https://archive.neon.kde.org/dev/unstable/pool/main/k/kirigami2 ${PKG} ${KIRIGAMI_VERSION}
done

...
```
You can also replace the other deb installations in dependencies.sh with this function, then all will look uniformly. Also, the length isn't more important than readability (if this function is used often, it'll beat the current code in length, though).